xbmc/cores/VideoPlayer/VideoPlayer.cpp

```diff
@@ -90,16 +90,14 @@ using namespace KODI::MESSAGING;
 void CSelectionStreams::Clear(StreamType type, StreamSource source)
 {
   CSingleLock lock(m_section);
-  for(int i=m_Streams.size()-1;i>=0;i--)
-  {
-    if(type && m_Streams[i].type != type)
-      continue;
-
-    if(source && m_Streams[i].source != source)
-      continue;
-
-    m_Streams.erase(m_Streams.begin() + i);
-  }
+  auto new_end =
+    std::remove_if(m_Streams.begin(),
+                   m_Streams.end(),
+                   [type, source](const SelectionStream &stream){
+                     return (type == STREAM_NONE || stream.type == type) &&
+                       (source == 0 || stream.source == source);
+                   });
```

Comment:
This can stay as is, but I think that readability is improved if newlines are single-indented by structuring these awkward lambas like this:

```c++
auto new_end = std::remove_if(m_Streams.begin(), m_Streams.end(),
  [type, source](const SelectionStream &stream)
  {
    return (type == STREAM_NONE || stream.type == type) &&
           (source == 0 || stream.source == source);
  });
```
other examples from a PR of mine: https://github.com/xbmc/xbmc/blob/master/xbmc/input/joysticks/generic/ButtonMapping.cpp#L141